shared/src/main/java/uk/gov/di/authentication/shared/services/mfa/MfaMethodsService.java

```diff
@@ -299,4 +307,83 @@ private boolean updateRequestChangesMethodType(
                 : !MFAMethodType.SMS.name().equals(methodTypeFromExisting);
     }
+
+    public Optional<MfaMigrationFailureReason> migrateMfaCredentialsForUser(String email) {
+        // Bail if user doesn't exist
+        Optional<UserProfile> maybeUserProfile = persistentService.getUserProfileFromEmail(email);
+        Optional<UserCredentials> maybeUserCredentials =
+                Optional.ofNullable(persistentService.getUserCredentialsFromEmail(email));
+        if (maybeUserProfile.isEmpty() || maybeUserCredentials.isEmpty())
+            return Optional.of(MfaMigrationFailureReason.NO_USER_FOUND_FOR_EMAIL);
+
+        UserProfile userProfile = maybeUserProfile.get();
+        UserCredentials userCredentials = maybeUserCredentials.get();
+
+        // Bail if already migrated
+        if (userProfile.getMfaMethodsMigrated())
+            return Optional.of(MfaMigrationFailureReason.ALREADY_MIGRATED);
+
+        // TODO - AUT-2198 - This method obfuscates things a bit
+        //  It includes only enabled UserCredentials and verified phone numbers. What should we do
+        // if they aren't either of these?
+        var maybeNonMigratedMfaMethod =
+                getMfaMethodForNonMigratedUser(userProfile, userCredentials);
+
+        // Bail if no MFA methods to migrate
+        if (maybeNonMigratedMfaMethod.isEmpty()) {
+            persistentService.setMfaMethodsMigrated(email, true);
+            return Optional.empty();
+            // TODO - AUT-2198 - By doing this, we need to make sure that all new MFA methods are
+            // getting added the new way before running before running a bulk migration, otherwise
+            // we could end up with a situation where we think something is migrated but actually
+            // isn't.
+            // TODO - AUT-2198 - Maybe we shouldn't do this given the two TODOs below
+        }
+
+        var nonMigratedMfaMethod = maybeNonMigratedMfaMethod.get();
+
+        return nonMigratedMfaMethod.method() instanceof AuthAppMfaDetail
+                ? migrateAuthAppToNewFormat(
+                        email,
+                        nonMigratedMfaMethod) // TODO - AUT-2198 - What should do for non-enabled
+                // auth app?
+                : migrateSmsToNewFormat(
+                        email,
+                        nonMigratedMfaMethod,
+                        userProfile); // TODO - AUT-2198 - What should do for non-verified phone?
+    }
+
+    private Optional<MfaMigrationFailureReason> migrateAuthAppToNewFormat(
+            String email, MfaMethodData mfaMethodData) {
+        var method = (AuthAppMfaDetail) mfaMethodData.method();
+        persistentService.migrateMfaMethodsToCredentialsTableForUser(
+                email,
+                MFAMethod.authAppMfaMethod(
+                        method.credential(),
+                        true,
+                        true,
+                        mfaMethodData.priorityIdentifier(),
```

Comment:
Don't think the non-migrated auth app will have a priorityIdentier as this is a new field for migrated items.  Set to 'default'?